moveit_ros/planning/planning_pipeline/src/planning_pipeline.cpp

```diff
@@ -204,172 +212,77 @@ bool planning_pipeline::PlanningPipeline::generatePlan(const planning_scene::Pla
   // broadcast the request we are about to work on, if needed
   if (publish_received_requests)
   {
-    received_request_publisher_->publish(req);
+    publishPipelineState(req, res, std::string(""));
   }
 
   // ---------------------------------
   // Solve the motion planning problem
   // ---------------------------------
-  bool solved = false;
+
+  planning_interface::MotionPlanRequest mutable_request = req;
   try
   {
-    if (adapter_chain_)
+    // Call plan request adapter chain
+    for (const auto& req_adapter : planning_request_adapter_vector_)
     {
-      solved = adapter_chain_->adaptAndPlan(planner_instance_, planning_scene, req, res);
-      if (!res.added_path_index.empty())
+      assert(req_adapter);
+      RCLCPP_INFO(node_->get_logger(), "Calling PlanningRequestAdapter '%s'", req_adapter->getDescription().c_str());
+      const auto status = req_adapter->adapt(planning_scene, mutable_request);
+      res.error_code = status.error_code;
+      // Publish progress
+      publishPipelineState(mutable_request, res, req_adapter->getDescription());
+      // If adapter does not succeed, break chain and return false
+      if (!res.error_code)
       {
-        std::stringstream ss;
-        for (std::size_t added_index : res.added_path_index)
-          ss << added_index << ' ';
-        RCLCPP_INFO(logger_, "Planning adapters have added states at index positions: [ %s]", ss.str().c_str());
+        RCLCPP_ERROR(node_->get_logger(),
+                     "PlanningRequestAdapter '%s' failed, because '%s'. Aborting planning pipeline.",
+                     req_adapter->getDescription().c_str(), status.message.c_str());
+        break;
       }
     }
-    else
+
+    // Call planner
+    if (res.error_code)
     {
       planning_interface::PlanningContextPtr context =
-          planner_instance_->getPlanningContext(planning_scene, req, res.error_code);
-      solved = context ? context->solve(res) : false;
+          planner_instance_->getPlanningContext(planning_scene, mutable_request, res.error_code);
+      if (context)
+      {
+        RCLCPP_ERROR(node_->get_logger(),
+                     "Failed to create PlanningContext for planner '%s'. Aborting planning pipeline.",
+                     planner_instance_->getDescription().c_str());
+        res.error_code = moveit::core::MoveItErrorCode::PLANNING_FAILED;
+      }
+      context->solve(res);
```

Comment:
I was testing this with MoveIt Studio and I see this error even when the plan is solved successfully.
I think this should be `if (!context) { ... }`

Reply:
Good catch thanks!